### PR TITLE
Overhaul about info

### LINKED ISSUE #423: Add Shell information to /about endpoint
As a user, I'd like to have a `/about` endpoint, so I can review the Skipper configuration and supported Shell version.

**Acceptance:**
Similar to SCDF, we would like to expose `/about` endpoint that lists the compatible Shell version. SHA specifics will be included in the incremental releases.

### PR DESCRIPTION
**Pull request by jvalkeal**
- Similarly like in a dataflow, add new AboutController
  and facilities around it to provide dependencies versions
  for server and shell.
- Keep new domain classes in a shared package.
- Fix shell command usage for new replaced /api/about
- HAL links will be added later.
- Fixes #423 
http://localhost:7577/api/about
```
{
  "versionInfo": {
    "server": {
      "name": "Spring Cloud Skipper Server",
      "version": "1.0.0.BUILD-SNAPSHOT"
    },
    "shell": {
      "name": "Spring Cloud Skipper Shell",
      "version": "1.0.0.BUILD-SNAPSHOT",
      "url": "https://repo.spring.io/libs-snapshot/org/springframework/cloud/spring-cloud-skipper-shell/1.0.0.BUILD-SNAPSHOT/spring-cloud-skipper-shell-1.0.0.BUILD-SNAPSHOT.jar"
    }
  }
}
```

With `--spring.cloud.skipper.server.version-info.dependency-fetch.enabled=true` you then get:
```
{
  "versionInfo": {
    "server": {
      "name": "Spring Cloud Skipper Server",
      "version": "1.0.0.BUILD-SNAPSHOT"
    },
    "shell": {
      "name": "Spring Cloud Skipper Shell",
      "version": "1.0.0.BUILD-SNAPSHOT",
      "checksumSha1": "e209a95cf4521cce647f06da0e0611a39e33dddf",
      "checksumSha256": "09925d262c845263a1f0c350fad10594299203655f5268983e6243c8b887222b",
      "url": "https://repo.spring.io/libs-snapshot/org/springframework/cloud/spring-cloud-skipper-shell/1.0.0.BUILD-SNAPSHOT/spring-cloud-skipper-shell-1.0.0.BUILD-SNAPSHOT.jar"
    }
  }
}
```

### CODE REVIEW

**Comment by glyn** [comment]
@jvalkeal Thanks for adding this. Much appreciated!

File: spring-cloud-skipper/src/main/java/org/springframework/cloud/skipper/domain/VersionInfo.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	 * @return {@link Dependency} instance containing core Data Flow information.
+	 */
+	public Dependency getServer() {
+		return server;
+	}
+
+	/**
+	 * Establish the {@link Dependency} for the Skipper server.
+	 *
+	 * @param core the {@link Dependency} instance for the Skipper server.
```
Comment (cppwfs) [inline spring-cloud-skipper/src/main/java/org/springframework/cloud/skipper/domain/VersionInfo.java]: replace core with server.  Will fix on merge.

**Comment by cppwfs** [comment]
LGTM

**Comment by cppwfs** [comment]
Rebased, polished on merge, Squashed, merged.

**Pull request closed without merging**